src/rules.ts

```diff
@@ -161,6 +161,9 @@ export class Rule implements IRule {
       case 'no_cache': {
         return `${this.name}-${Math.random()}`
       }
+      default: {
+        return `${this.name}-${this.cache(parent, args)}`
```

Comment:
Can `this.cache` be null or non-function?

Reply:
it cannot, if cache option is falsy it gets set to contextual